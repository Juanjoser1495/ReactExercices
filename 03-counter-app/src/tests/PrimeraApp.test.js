import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimerApp";

describe("Pruebas en <PrimeraApp />", () => {
  //   test("Debe de mostrar el mensaje 'Hola Soy Goku'", () => {
  //     const saludo = "Hola Soy Goku";
  //     const { getByText } = render(<PrimeraApp saludo={saludo}></PrimeraApp>);
  //     expect(getByText(saludo)).toBeInTheDocument();
  //   });

  test("Debe de mostrar el <PrimeraApp> correctamnete", () => {
    const saludo = "Hola Soy Goku";
    const wrapper = shallow(<PrimeraApp saludo={saludo}></PrimeraApp>);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el subtitulo enviado por props", () => {
    const saludo = "Hola Soy Goku";
    const subtitulo = "Soy un subtitulo";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo}></PrimeraApp>
    );
    const returned = wrapper.find("h3").text();
    expect(returned).toBe(subtitulo);
  });
});
