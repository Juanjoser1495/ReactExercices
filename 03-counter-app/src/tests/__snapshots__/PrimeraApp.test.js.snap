// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar el <PrimeraApp> correctamnete 1`] = `
<Fragment>
  <h1>
    Hola Mundo const
  </h1>
  <h2>
    Hola Soy Goku
  </h2>
  <h3>
    Soy un subtitulo
  </h3>
  <p>
    Mi primera app
  </p>
</Fragment>
`;
